openmmtools/tests/test_cache.py

```diff
@@ -221,6 +221,25 @@ def test_get_incompatible_context(self):
         assert len(cache) == 4
         assert n_contexts == 4
 
+    def test_get_context_any_integrator(self):
+        """ContextCache.get_context first search the cache when integrator is unspecified."""
+        cache = ContextCache()
+        state1, state2 = self.incompatible_states[:2]
+
+        # First we create a Context in state1.
+        cache.get_context(state1, copy.deepcopy(self.verlet_2fs))
+        assert len(cache) == 1
```

Comment:
Oh, sorry, that's just a particular test case. We instantiate a new local, empty `ContextCache` above (I'm not using the global cache in this test), so that `assert` is just checking that the cache (originally empty) has indeed created and stored a `Context` object after the call to `get_context`.

If we were using the global cache in `cache.global_context_cache` that could be 20 depending on what has happened before.